### PR TITLE
CodeV16p1

### PR DESCRIPTION
**Pull request by tvami**
* move a lot of `double`-s to `float`-s
 * create fillDescribtion, with that get rid of the cff content
 * introduces eras in the cff and modify constants based on eras
 * rename `BS_Pterr` to `BS_PterrOverPt` and add `BS_PterrOverPt2`
 * add cutflow histogram having all cuts in a single plot, the order here should be revised

### CODE REVIEW

**Review by dapparu — COMMENTED** [review]
I'm not sure passing track instead hscp in preselection is a good thing. We lose some informations which could be helpful in the future, like timing info.

**Review by dapparu — APPROVED** [review]

File: Analyzer/interface/TupleMaker.h [hunk 1]
```diff
@@ -206,17 +206,17 @@ void TupleMaker::initializeTuple(Tuple *&tuple,
                                  bool isSignal,
                                  unsigned int NCuts,
                                  unsigned int NCuts_Flip,
-                                 double PtHistoUpperBound,
-                                 double MassHistoUpperBound,
+                                 float PtHistoUpperBound,
```
Comment (enibigir) [inline Analyzer/interface/TupleMaker.h:209]: @tvami 
By construction, min and max bins of TH1  are double. I am wondering if we could not keep those lower/upper edges as double-s?

**Review by enibigir — COMMENTED** [review]
May be take this occasion to rename "*_Im" to "*_Ih".
Let me what you think.
I checked the compilation  it's fine.

**Comment by tvami** [comment]
@dapparu 
> I'm not sure passing track instead hscp in preselection is a good thing. We lose some informations which could be helpful in the future, like timing info.

Ok, but there was a lot of duplicated code in the original, if we need the timing info maybe we can pass that as a separate argument in the future

File: Analyzer/interface/TupleMaker.h (hunk 1 above)
Comment (tvami) [reply Analyzer/interface/TupleMaker.h:209]: but we really dont need double-s, these numbers are not even revant upto one digit after the 0

**Comment by tvami** [comment]
> May be take this occasion to rename "__Im" to "__Ih". Let me what you think. I checked the compilation it's fine.

Sure, I can do that in a follow-up PR

**Pull request merged**